addons/crm/models/crm_lead.py

```diff
@@ -1164,57 +1164,74 @@ def get_rainbowman_message(self):
-        elif query_result['max_team_30'] == self.expected_revenue:
-            message = _('Boom! Team record for the past 30 days.')
-        elif query_result['max_team_7'] == self.expected_revenue:
-            message = _('Yeah! Deal of the last 7 days for the team.')
-        elif query_result['max_user_30'] == self.expected_revenue:
-            message = _('You just beat your personal record for the past 30 days.')
-        elif query_result['max_user_7'] == self.expected_revenue:
-            message = _('You just beat your personal record for the past 7 days.')
-        return message
+        if query_result['count_user_closed_year'] == 1:
+            return _('Go, go, go! Congrats for your first deal.')
+        elif self.expected_revenue and query_result['max_team_30'] == self.expected_revenue:
+            return _('Boom! Team record for the past 30 days.')
+        elif self.expected_revenue and query_result['max_team_7'] == self.expected_revenue:
+            return _('Yeah! Deal of the last 7 days for the team.')
+        elif self.expected_revenue and query_result['max_user_30'] == self.expected_revenue:
+            return _('You just beat your personal record for the past 30 days.')
+        elif self.expected_revenue and query_result['max_user_7'] == self.expected_revenue:
+            return _('You just beat your personal record for the past 7 days.')
+        elif query_result['count_user_closed_today'] == 5:
+            return _('You\'re on fire! Fifth deal won today 🔥')
+        elif query_result['count_country_closed_year'] == 1 and self.country_id:
+            return _('You just expanded the map! First win in %s.', self.country_id.name)
+        elif query_result['count_source_closed_year'] == 1 and self.source_id:
+            return _('Yay, your first win from %s!', self.source_id.name)
+        # checked here as it is its position in the priority order
+        elif len(self.message_ids) >= 25:
+            return _('Phew, that took some effort — but you nailed it. Good job!')
+        elif query_result['count_user_closed_today'] == 1 and query_result['count_user_closed_yesterday'] and query_result['count_user_closed_minus2day'] and not query_result['count_user_closed_minus3day']:
+            return _('You\'re on a winning streak. 3 deals in 3 days, congrats!')
+        elif query_result['min_day_close_30'] == self.day_close and len([duration for duration in self.duration_tracking.values() if duration >= 60]) >= 1:
+            return _('Wow, that was fast. That deal didn’t stand a chance!')
+        # use duration tracking field to determine if the task jumped from first to last stage
+        elif len([duration for duration in self.duration_tracking.values() if duration >= 60]) == 1:
+            first_stage = self.env['crm.stage'].search(
+                [('id', 'in', [int(stage_id) for stage_id, duration in self.duration_tracking.items() if duration >= 60])],
+                limit=1
+            )
```

Comment:
trivia but..

there's an operator that lets you evaluate and assign at the same time :)
avoids having to do it twice.
`elif len(stage_ids := [int(stage_id) for stage_id, duration in self.duration_tracking.items() if duration >= 60])]) == 1:`

 Also note if you just need to know if an iterator has "at least 1" of anything you can use `any`, so it stops early. Can be useful for the previous conditional block.